src/main/java/org/broadleafcommerce/vendor/amazon/s3/S3FileServiceProvider.java

```diff
@@ -80,15 +90,42 @@ public File getResource(String name) {
 
     @Override
     public File getResource(String name, FileApplicationType fileApplicationType) {
-        File returnFile = blFileService.getLocalResource(buildResourceName(name));
+        File returnFile = blFileService.getLocalResource(getResourceName(name));
         OutputStream outputStream = null;
         InputStream inputStream = null;
 
         try {
             S3Configuration s3config = s3ConfigurationService.lookupS3Configuration();
             AmazonS3Client s3 = getAmazonS3Client(s3config);
-            S3Object object = s3.getObject(new GetObjectRequest(s3config.getDefaultBucketName(), buildResourceName(name)));
-            
+            String resourceName = getResourceName(name);
```

Comment:
line 100 is an attempt to resolve the resource name on the local filesystem. If the resource name is found, then we know exactly what resource name to query the S3 file server for. If the resource path can not be found on the local filesystem, then the `getResourceName()` function will return null ( indicating that no resource was found). It is only at this point that the system will gather all possible resource names and query the E3 file server with all possible until it finds a match.